.github/workflows/update-automation.yaml

```
@@ -6,7 +6,66 @@ on:
   workflow_dispatch:
 
 jobs:
-  run:
+  update-automation:
+    name: Run Automation Tasks
     runs-on: ubuntu-latest
+    permissions:
+      contents: write
     steps:
-      - run: echo "Scheduled job to run automation tasks"
\ No newline at end of file
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+          
+      - name: Check if update needed
+        run: |
+          cd third-party-src
+          git fetch --tags
+          CURRENT_TAG=$(git describe --tags --exact-match HEAD 2>/dev/null | head -1)
+          if [ -z "$CURRENT_TAG" ]; then
+            echo "Error: Submodule is not on a tagged commit"
+            exit 1
+          fi
+          cd ..
+          
+          LATEST_TAG=$(curl -L \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            https://api.github.com/repos/microsoft/vscode/releases/latest | jq -r '.tag_name')
```

Nit: GitHub CLI should be available in GitHub Action environments, so this could be also written more concisely:
```suggestion
          LATEST_TAG=$(gh api repos/microsoft/vscode/releases/latest --template '{{.tag_name}}')
```